icu4c/source/common/unicode/uset.h

```diff
@@ -1756,26 +1759,28 @@ class USetElementIterator {
         return uset == other.uset && c == other.c && index == other.index;
     }
 
-    /** @draft ICU 76 */
+    /** @draft ICU 77 */
     bool operator!=(const USetElementIterator &other) const { return !operator==(other); }
 
-    /** @draft ICU 76 */
-    UnicodeString operator*() const {
+    /** @draft ICU 77 */
+    std::u16string operator*() const {
         if (c >= 0) {
-            return UnicodeString(c);
+            return c <= 0xffff ?
+                std::u16string({static_cast<char16_t>(c)}) :
+                std::u16string({U16_LEAD(c), U16_TRAIL(c)});
         } else if (index < totalCount) {
             int32_t length;
             const UChar *uchars = uset_getString(uset, index - rangeCount, &length);
             // assert uchars != nullptr;
-            return UnicodeString(uchars, length);
+            return {ConstChar16Ptr(uchars), static_cast<uint32_t>(length)};
```

Comment:
Shouldn't that be `std::u16string::size_type` instead of `uint32_t`?

Reply:
i prefer just casting to the unsigned version of the type and letting the constructor do the rest

if anything i would cast to size_t -- i find the type-specific something::size_type super clunky and unnecessary

Reply:
Since we need to have a cast anyway, I would prefer casting to the right thing directly. I agree that size_t is more readable though (and it follows from https://eel.is/c++draft/string.classes#general-1, https://eel.is/c++draft/string.classes#basic.string.general-3, and https://eel.is/c++draft/default.allocator#general-1 that `std::u16string::size_type` is `std::size_t`).

Reply:
fine, will do